perceval/components/source.py

```diff
@@ -21,66 +21,129 @@
 # SOFTWARE.
 
 import math
+import numpy as np
 
 from perceval.utils import SVDistribution, StateVector
-from typing import Dict
+from typing import Dict, Literal
 
 
 class Source:
-    r"""Definition of a single photon source. The default source is a perfect source (emits exactly 1 indistinguishable
-    photon each period).
-
-    :param brightness: the brightness of the source defined as the percentage of unique photon generation
-    :param purity: the ratio of time when photon is emitted alone
-    :param purity_model: `"random"` if additional photons are distinguishable, `"indistinguishable"` otherwise
-    :param indistinguishability: indistinguishability parameter as defined by `indistinguishability_model`
-    :param indistinguishability_model: `"homv"` defines indistinguishability as HOM visibility, `"linear"` defines
-        indistinguishability as ratio of indistinguishable photons
-    :param context: gives a local context for source specific features, like `"discernability_tag"`
-    """
     def __init__(self, brightness: float = 1,
-                 purity: float = 1,
-                 purity_model: str = "random",  # Literal["random", "indistinguishable"]
+                 multiphoton_component: float = None,
+                 multiphoton_model: Literal["distinguishable", "indistinguishable"] = "distinguishable",
+                 purity: float = None,
                  indistinguishability: float = 1,
-                 indistinguishability_model: str = "homv",  # Literal["homv", "linear"]
+                 indistinguishability_model: Literal["homv", "linear"] = "homv",
+                 overall_transmission: float = 1,
                  context: Dict = None) -> None:
+        r"""Definition of a source
+
+        :param brightness: the probability per laser pulse to emmit at least one photon. Indenpendent of all losses.
+        :param multiphoton_component: second order intensity autocorrelation at zero time delay :math:`g^{(2)}(0)`
+        :param multiphoton_model: `distinguishable` if additional photons are distinguishable, `indistinguishable`
+          otherwise
+        :param purity: preserved for back-compatibility if multiphoton_model is not set.`
+        :param indistinguishability: indistinguishability parameter as defined by `indistinguishability_model`
+        :param indistinguishability_model: `homv` defines indistinguishability as 2-photon wavepacket overlap,
+            `linear` defines indistinguishability as ratio of indistinguishable photons
+        :param overall_transmission: transmission of the optical system.
```

Comment:
How can a user pick a good value for `overall_transmission`? Is it really a Source parameter or should it be a Processor parameter?